### PR TITLE
Tiles renames and documentation

### PR DESCRIPTION
**Pull request by groud**
This PR is mainly fills in the documentation for tiles-related stuff. However, while working on it, I spotted a bug and a few things to rename, so I put it into the PR too. 

~~This is a draft for now as I need https://github.com/godotengine/godot/pull/52277 to be merged first.~~

### CODE REVIEW

File: doc/classes/TileData.xml [hunk 1]
```diff
@@ -11,70 +11,81 @@
...
 			<description>
+				Returns the tile's terrain bit for the given [code]peering_bit[/code] direction.
 			</description>
 		</method>
 		<method name="is_collision_polygon_one_way" qualifiers="const">
 			<return type="bool" />
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="polygon_index" type="int" />
 			<description>
+				Returns if one-way collisions are enabled for the polygon at index [code]polygon_index[/code] for TileSet physics layer with index [code]layer_id[/code].
```
Comment (mhilbrunner) [inline doc/classes/TileData.xml:80]:
```suggestion
				Returns whether one-way collisions are enabled for the polygon at index [code]polygon_index[/code] for TileSet physics layer with index [code]layer_id[/code].
```

File: doc/classes/TileData.xml [hunk 2]
```diff
@@ -99,59 +112,55 @@
 			<argument index="1" name="polygon_index" type="int" />
 			<argument index="2" name="polygon" type="PackedVector2Array" />
 			<description>
+				Set the points of the polygon at index [code]polygon_index[/code] for TileSet physics layer with index [code]layer_id[/code].
```
Comment (mhilbrunner) [inline doc/classes/TileData.xml:115]:
```suggestion
				Sets the points of the polygon at index [code]polygon_index[/code] for TileSet physics layer with index [code]layer_id[/code].
```

File: doc/classes/TileData.xml [hunk 3]
```diff
@@ -99,59 +112,55 @@
...
 			</description>
 		</method>
 		<method name="set_collision_polygons_count">
 			<return type="void" />
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="polygons_count" type="int" />
 			<description>
+				Set the polygons count for TileSet physics layer with index [code]layer_id[/code].
```
Comment (mhilbrunner) [inline doc/classes/TileData.xml:123]:
```suggestion
				Sets the polygons count for TileSet physics layer with index [code]layer_id[/code].
```

File: doc/classes/TileData.xml [hunk 4]
```diff
@@ -99,59 +112,55 @@
...
 			</description>
 		</method>
 		<method name="set_custom_data">
 			<return type="void" />
 			<argument index="0" name="layer_name" type="String" />
 			<argument index="1" name="value" type="Variant" />
 			<description>
+				Set the tile's custom data value for the TileSet custom data layer with name [code]layer_name[/code].
```
Comment (mhilbrunner) [inline doc/classes/TileData.xml:131]:
```suggestion
				Sets the tile's custom data value for the TileSet custom data layer with name [code]layer_name[/code].
```

File: doc/classes/TileData.xml [hunk 5]
```diff
@@ -99,59 +112,55 @@
...
 			</description>
 		</method>
 		<method name="set_custom_data_by_layer_id">
 			<return type="void" />
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="value" type="Variant" />
 			<description>
+				Set the tile's custom data value for the TileSet custom data layer with index [code]layer_id[/code].
```
Comment (mhilbrunner) [inline doc/classes/TileData.xml:139]:
```suggestion
				Sets the tile's custom data value for the TileSet custom data layer with index [code]layer_id[/code].
```

File: doc/classes/TileData.xml [hunk 6]
```diff
@@ -99,59 +112,55 @@
...
 			</description>
 		</method>
 		<method name="set_navigation_polygon">
 			<return type="void" />
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="navigation_polygon" type="NavigationPolygon" />
 			<description>
+				Set the navigation polygon for the TileSet navigation layer with index [code]layer_id[/code].
```
Comment (mhilbrunner) [inline doc/classes/TileData.xml:147]:
```suggestion
				Sets the navigation polygon for the TileSet navigation layer with index [code]layer_id[/code].
```

File: doc/classes/TileData.xml [hunk 7]
```diff
@@ -99,59 +112,55 @@
...
 			</description>
 		</method>
 		<method name="set_occluder">
 			<return type="void" />
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="occluder_polygon" type="OccluderPolygon2D" />
 			<description>
+				Set the occluder for the TileSet occlusion layer with index [code]layer_id[/code].
```
Comment (mhilbrunner) [inline doc/classes/TileData.xml:155]:
```suggestion
				Sets the occluder for the TileSet occlusion layer with index [code]layer_id[/code].
```

File: doc/classes/TileSetScenesCollectionSource.xml [hunk 8]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="TileSetScenesCollectionSource" inherits="TileSetSource" version="4.0">
 	<brief_description>
+		Exposes a set of scenes as tiles for a TileSet resource.
```
Comment (mhilbrunner) [inline doc/classes/TileSetScenesCollectionSource.xml:4]:
```suggestion
		Exposes a set of scenes as tiles for a [TileSet] resource.
```

**Review by mhilbrunner — CHANGES_REQUESTED** [review]
Looks good. A few nitpicks below. :)

**Comment by groud** [comment]
Alright, I addressed the feedback and added the missing documentation for TileSetScenesCollectionSource I forgot.
Let me know if anything is still missing :).

**Review by mhilbrunner — APPROVED** [review]
Looks good now. :)

**Review by KoBeWi — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks!

**Pull request merged**